datalad/crawler/dbs/base.py

```diff
@@ -85,7 +87,15 @@ def save(self):
         lgr.debug("Writing %s to %s" % (self.__class__.__name__, self._filepath))
         with open(self._filepath, 'w') as f:
             json.dump(db, f, indent=2, sort_keys=True, separators=(',', ': '))
-        self.repo.git_add(self._filepath)  # stage to be committed
+
+        # stage to be committed:
+        if isinstance(self.repo, AnnexRepo):
+            self.repo.add(self._filepath, git=True)
+        elif isinstance(self.repo, GitRepo):
+            self.repo.add(self._filepath)
+        else:
+            raise ValueError("Unknown repo: %s" % self.repo)
```

Comment:
Well, the original idea of just calling `add` via `super` doesn't work in case of dynamic usage. So, yes we need it. This or the just killed git_ - prefix.

Do you remember the **kwargs - idea, passed to any git call for custom options? If we provide this for all the methods, we could always pass `git=True` and have the assertion inside GitRepo's method (that it is True, if it is there) without the explicit declaration of this option.
What do you think about that approach? A little bit less ugly or not?